src/ButtonCatalog.js

```diff
@@ -31,22 +37,68 @@ export class ButtonHero extends Component {
 
   render() {
     return (
-      <div>
-        <button className='hero-button mdc-button' ref={this.initRipple}>
-          Text
-        </button>
-        <button className='hero-button mdc-button mdc-button--raised' ref={this.initRipple}>
-          Raised
-        </button>
-        <button className='hero-button mdc-button mdc-button--unelevated' ref={this.initRipple}>
-          Unelevated
-        </button>
-        <button className='hero-button mdc-button mdc-button--outlined' ref={this.initRipple}>
-          Outlined
-        </button>
-      </div>
+        <div>
+          <button className='hero-button mdc-button' ref={this.initRipple}>
+            Text
+          </button>
+          <button className='hero-button mdc-button mdc-button--raised'
+                  ref={this.initRipple}>
+            Raised
+          </button>
+          <button className='hero-button mdc-button mdc-button--unelevated'
+                  ref={this.initRipple}>
+            Unelevated
+          </button>
+          <button className='hero-button mdc-button mdc-button--outlined'
+                  ref={this.initRipple}>
+            Outlined
+          </button>
+        </div>
+    );
+  }
+
+}
+
+export class ButtonHero extends Component {
+  type = {
+    text: '',
+    raised: 'mdc-button--raised',
+    unelevated: 'mdc-button--unelevated',
+    outlined: 'mdc-button--outlined',
+  };
+
+  label = 'Button Text';
+
+  constructor(props) {
+    super(props);
+    this.ripple = null;
+    this.buttonRef = React.createRef();
+  }
+
+  componentWillUnmount() {
+    if (this.ripple) this.ripple.destroy();
+  }
+
+  componentDidUpdate() {
+    if (this.ripple) this.ripple.destroy();
+    this.ripple = new MDCRipple(this.buttonRef.current);
```

Comment:
When the component updates to a different style the ripple is broken. Most components will need something like this since we do a lot of setup when initialized.

Reply:
gotcha